### PR TITLE
[pkgcfg] Add nginx config

### PR DESCRIPTION
**Pull request by mikeland86**
## Summary

Adds basic nginx config. Creates a simple config file and a shim that uses it. Can be customized with environment variable.

Note that the readme is currently not implemented but I'm adding so it will be available once it is.

## How was it tested?

```
devbox add nginx
devbox shell
nginx
```

### CODE REVIEW

File: pkgcfg/package-configuration/nginx.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "nginx",
+  "version": "0.0.1",
+  "readme": "To start nginx use command \"nginx\". nginx is configured to use .devbox/conf/nginx.conf. Use $NGINX_CONFDIR to change the configuration directory. Use $NGINX_LOGDIR to change the log directory. Use $NGINX_PIDDIR to change the pid directory. Use $NGINX_RUNDIR to change the run directory. Use $NGINX_SITESDIR to change the sites directory. Use $NGINX_TMPDIR to change the tmp directory. Use $NGINX_USER to change the user. Use $NGINX_GROUP to customize.",
```
Comment (savil) [inline pkgcfg/package-configuration/nginx.json:4]: Does this Readme get rendered somewhere or is it for reading within this file?

Would suggest adding some new line chars inside this

File: pkgcfg/package-configuration/nginx/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+events {}
+http{
+server {
+         listen       80;
+         listen       [::]:80;
```
Comment (savil) [inline pkgcfg/package-configuration/nginx/nginx.conf:5]: Why the two separate listen calls?

File: pkgcfg/pkgcfg.go [hunk 3]
```diff
@@ -55,6 +55,7 @@ func CreateFiles(pkg, rootDir string) error {
 		}
 		var buf bytes.Buffer
 		if err = t.Execute(&buf, map[string]string{
+			"UserRoot":             rootDir,
```
Comment (savil) [inline pkgcfg/pkgcfg.go:58]: Is root dir the absolute path to project directory?

**Review by savil — APPROVED** [review]
Nice!

File: pkgcfg/package-configuration/nginx.json (hunk 1 above)
Comment (mikeland86) [reply pkgcfg/package-configuration/nginx.json:4]: Currently readme is not used at all. I'm planning on implementing it and improving them pending a conversation with @loreto

File: pkgcfg/package-configuration/nginx/nginx.conf (hunk 2 above)
Comment (mikeland86) [reply pkgcfg/package-configuration/nginx/nginx.conf:5]: Not sure, copy pasted this from @Lagoja 's template

File: pkgcfg/pkgcfg.go (hunk 3 above)
Comment (mikeland86) [reply pkgcfg/pkgcfg.go:58]: yep. I named it UserRoot because we also have DevboxRoot which is the path to `.devbox`

Comment (savil) [reply pkgcfg/pkgcfg.go:58]: we also have the concept of `configDir` which is the path to where the project's `devbox.json` is kept. 

ugh, although this `configDir` is written as `Devbox.srcDir` in `devbox.go`. I can send a PR to update that.

**Pull request merged**